From 0f6462f7df5d2b22c180e1bb02190a89fc64435f Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Wed, 5 Nov 2025 09:28:03 -0800
Subject: [PATCH 2/3] add gfx1103 iGPU support

- add gfx1103 iGPU support
- gfx1102 was missing from the is_rdna method
- sort the gpu-lists to be in numerical order

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt                   | 2 +-
 include/aotriton/util.h          | 1 +
 tritonsrc/attn_torch_function.py | 2 +-
 v3python/compile.py              | 2 +-
 v3python/gpu_targets.py          | 5 +++++
 v3src/util.cc                    | 2 ++
 6 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c31b12c..635c9a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,7 @@ option(AOTRITON_INHERIT_SYSTEM_SITE_TRITON "Use system site packages and verify
 set(AOTRITON_USE_LOCAL_TRITON_WHEEL "" CACHE STRING "Substitute install from third_party/triton with install from local pip wheel package.")
 set(AOTRITON_GPU_BUILD_TIMEOUT "8.0" CACHE STRING "GPU kernel compiler times out after X minutes. 0 for indefinite. Highly recommended if AOTRITON_BUILD_FOR_TUNING=On.")
 option(AOTRITON_TERMINATE_WHEN_GPU_BUILD_TIMEOUT "compile.py will return error when GPU built timeout. Defensive option for CI" OFF)
-set(AOTRITON_TARGET_ARCH "gfx90a;gfx942;gfx950;gfx1100;gfx1101;gfx1102;gfx1151;gfx1150;gfx1201;gfx1200" CACHE STRING "Target GPU Architecture. Select all GPUs within the given list")
+set(AOTRITON_TARGET_ARCH "gfx90a;gfx942;gfx950;gfx1100;gfx1101;gfx1102;gfx1103;gfx1151;gfx1150;gfx1201;gfx1200" CACHE STRING "Target GPU Architecture. Select all GPUs within the given list")
 set(TARGET_GPUS "OBSOLETE" CACHE STRING "OBSOLETE. To select only one GPU, use AOTRITON_TARGET_ARCH or AOTRITON_OVERRIDE_TARGET_GPUS.")
 set(AOTRITON_OVERRIDE_TARGET_GPUS "" CACHE STRING "Override AOTRITON_TARGET_ARCH, and only build for GPUs within this list.")
 set(AOTRITON_ALT_TRITON_WHEEL_CONFIG_FILE "" CACHE STRING "Specify how to use alternative Triton wheels for different architectures in a YAML config file.")
diff --git a/include/aotriton/util.h b/include/aotriton/util.h
index 4ed494f..accc2c8 100644
--- a/include/aotriton/util.h
+++ b/include/aotriton/util.h
@@ -59,6 +59,7 @@ enum AOTRITON_API Gpu : uint64_t {
   GPU_AMD_ARCH_GFX1100_MOD0 = TRICAT(GpuVendor::kAMD, 0x1100, 0),
   GPU_AMD_ARCH_GFX1101_MOD0 = TRICAT(GpuVendor::kAMD, 0x1101, 0),
   GPU_AMD_ARCH_GFX1102_MOD0 = TRICAT(GpuVendor::kAMD, 0x1102, 0),
+  GPU_AMD_ARCH_GFX1103_MOD0 = TRICAT(GpuVendor::kAMD, 0x1103, 0),
   GPU_AMD_ARCH_GFX1151_MOD0 = TRICAT(GpuVendor::kAMD, 0x1151, 0),
   GPU_AMD_ARCH_GFX1150_MOD0 = TRICAT(GpuVendor::kAMD, 0x1150, 0),
   GPU_AMD_ARCH_GFX950_MOD0  = TRICAT(GpuVendor::kAMD,  0x950, 0),
diff --git a/tritonsrc/attn_torch_function.py b/tritonsrc/attn_torch_function.py
index 0b7f7cd..c798e8c 100644
--- a/tritonsrc/attn_torch_function.py
+++ b/tritonsrc/attn_torch_function.py
@@ -34,7 +34,7 @@ def evaluate_gfx_arch_within(arch_list):
     return any(arch in gcn_arch_name for arch in arch_list)
 
 def is_rdna():
-    return evaluate_gfx_arch_within(['gfx1100', 'gfx1101', 'gfx1200', 'gfx1201', 'gfx1030', 'gfx1150', 'gfx1151'])
+    return evaluate_gfx_arch_within(['gfx1100', 'gfx1101', 'gfx1102', 'gfx1103', 'gfx1150', 'gfx1151', 'gfx1200', 'gfx1201'])
 
 IS_RDNA = is_rdna()
 
diff --git a/v3python/compile.py b/v3python/compile.py
index f72ddf2..014140e 100644
--- a/v3python/compile.py
+++ b/v3python/compile.py
@@ -15,7 +15,7 @@ import json
 from triton.backends.compiler import GPUTarget
 
 KNOWN_TARGETS_64 = ['gfx90a', 'gfx942', 'gfx950']
-KNOWN_TARGETS_32 = ['gfx1100', 'gfx1101', 'gfx1102', 'gfx1201', 'gfx1200', 'gfx1151', 'gfx1150', 'gfx1250']
+KNOWN_TARGETS_32 = ['gfx1100', 'gfx1101', 'gfx1102', 'gfx1103', 'gfx1150', 'gfx1151', 'gfx1200', 'gfx1201', 'gfx1250']
 
 KNOWN_TARGETS = {
   arch : GPUTarget('hip', arch, 64) for arch in KNOWN_TARGETS_64
diff --git a/v3python/gpu_targets.py b/v3python/gpu_targets.py
index 8492757..a9387da 100644
--- a/v3python/gpu_targets.py
+++ b/v3python/gpu_targets.py
@@ -12,6 +12,7 @@ AOTRITON_SUPPORTED_GPUS = (
     'gfx1100_mod0',
     'gfx1101_mod0',
     'gfx1102_mod0',
+    'gfx1103_mod0',
     'gfx1151_mod0',
     'gfx1150_mod0',
     'gfx1201_mod0',
@@ -24,6 +25,7 @@ AOTRITON_SUPPORTED_GPUS = (
 AOTRITON_TUNING_DATABASE_REUSE = {
     'gfx1101_mod0' : 'gfx1100_mod0',
     'gfx1102_mod0' : 'gfx1100_mod0',
+    'gfx1103_mod0' : 'gfx1100_mod0',
     'gfx1200_mod0' : 'gfx1201_mod0',
     'gfx1151_mod0' : 'gfx1100_mod0',
     'gfx1150_mod0' : 'gfx1100_mod0',
@@ -37,6 +39,7 @@ AOTRITON_ARCH_TO_PACK = {
     'gfx1100'   : 'gfx11xx',
     'gfx1101'   : 'gfx11xx',
     'gfx1102'   : 'gfx11xx',
+    'gfx1103'   : 'gfx11xx',
     'gfx1151'   : 'gfx11xx',
     'gfx1150'   : 'gfx11xx',
     'gfx1201'   : 'gfx120x',
@@ -55,6 +58,7 @@ AOTRITON_ARCH_WARPSIZE = {
     'gfx1100'    : 32,
     'gfx1101'    : 32,
     'gfx1102'    : 32,
+    'gfx1103'    : 32,
     'gfx1151'    : 32,
     'gfx1150'    : 32,
     'gfx1201'    : 32,
@@ -69,6 +73,7 @@ AOTRITON_ARCH_PRODUCTION_LINE = {
     'gfx1100'    : 'RDNA',
     'gfx1101'    : 'RDNA',
     'gfx1102'    : 'RDNA',
+    'gfx1103'    : 'RDNA',
     'gfx1151'    : 'RDNA',
     'gfx1150'    : 'RDNA',
     'gfx1201'    : 'RDNA',
diff --git a/v3src/util.cc b/v3src/util.cc
index f70804e..0ed58b6 100644
--- a/v3src/util.cc
+++ b/v3src/util.cc
@@ -51,6 +51,7 @@ std::unordered_map<std::string, GpuClassifier> LazyGpu::string_to_classifier = {
   {"gfx1100", DummyClassifier<GPU_AMD_ARCH_GFX1100_MOD0>() },
   {"gfx1101", DummyClassifier<GPU_AMD_ARCH_GFX1101_MOD0>() },
   {"gfx1102", DummyClassifier<GPU_AMD_ARCH_GFX1102_MOD0>() },
+  {"gfx1103", DummyClassifier<GPU_AMD_ARCH_GFX1103_MOD0>() },
   { "gfx950", DummyClassifier<GPU_AMD_ARCH_GFX950_MOD0 >() },
   {"gfx1150", DummyClassifier<GPU_AMD_ARCH_GFX1150_MOD0>() },
   {"gfx1151", DummyClassifier<GPU_AMD_ARCH_GFX1151_MOD0>() },
@@ -78,6 +79,7 @@ bool isArchExperimentallySupported(hipStream_t stream) {
           vendor_arch == CAT32(GpuVendor::kAMD, 0x1100) ||
           vendor_arch == CAT32(GpuVendor::kAMD, 0x1101) ||
           vendor_arch == CAT32(GpuVendor::kAMD, 0x1102) ||
+          vendor_arch == CAT32(GpuVendor::kAMD, 0x1103) ||
           vendor_arch == CAT32(GpuVendor::kAMD, 0x1200) ||
           vendor_arch == CAT32(GpuVendor::kAMD, 0x1250));
 }
-- 
2.43.0

